src/main/webapp/app/exercises/shared/exercise-scores/exercise-scores-export-button.component.ts

```diff
@@ -3,68 +3,123 @@ import { roundValueSpecifiedByCourseSettings } from 'app/shared/util/utils';
 import { faDownload } from '@fortawesome/free-solid-svg-icons';
 import { ExportToCsv } from 'export-to-csv';
+import { TestCaseResult } from 'app/entities/test-case-result.model';
 
 @Component({
     selector: 'jhi-exercise-scores-export-button',
     template: `
-        <button class="btn btn-info btn-sm me-1" (click)="exportResults()">
+        <div *ngIf="isProgrammingExerciseResults" ngbDropdown class="d-inline">
+            <button id="export-results-dropdown" class="btn btn-info btn-sm me-1" ngbDropdownToggle>
+                <fa-icon [icon]="faDownload"></fa-icon>
+                <span class="d-none d-md-inline" jhiTranslate="artemisApp.exercise.exportResults.title">Export Results</span>
+            </button>
+            <div ngbDropdownMenu="export-results-dropdown">
+                <button ngbDropdownItem (click)="exportResults(false, false)" jhiTranslate="artemisApp.exercise.exportResults.withoutTestCases">Without test cases</button>
+                <button
+                    ngbTooltip="{{ 'artemisApp.exercise.exportResults.withTestCasesTooltip' | artemisTranslate }}"
+                    placement="left"
+                    ngbDropdownItem
+                    (click)="exportResults(true, false)"
+                    jhiTranslate="artemisApp.exercise.exportResults.withTestCases"
+                >
+                    With test cases
+                </button>
+                <button
+                    ngbTooltip="{{ 'artemisApp.exercise.exportResults.withTestCasesAndFeedbackTooltip' | artemisTranslate }}"
+                    placement="left"
+                    ngbDropdownItem
+                    (click)="exportResults(true, true)"
+                    jhiTranslate="artemisApp.exercise.exportResults.withTestCasesAndFeedback"
+                >
+                    With test cases<br />and feedback
+                </button>
+            </div>
+        </div>
+        <button *ngIf="!isProgrammingExerciseResults" class="btn btn-info btn-sm me-1" (click)="exportResults(false, false)">
             <fa-icon [icon]="faDownload"></fa-icon>
-            <span class="d-none d-md-inline" jhiTranslate="artemisApp.exercise.exportResults">Export Results</span>
+            <span class="d-none d-md-inline" jhiTranslate="artemisApp.exercise.exportResults.title">Export Results</span>
         </button>
     `,
 })
 @Injectable({ providedIn: 'root' })
-export class ExerciseScoresExportButtonComponent {
+export class ExerciseScoresExportButtonComponent implements OnInit {
     @Input() exercises: Exercise[] = []; // Used to export multiple scores together
     @Input() exercise: Exercise | ProgrammingExercise;
 
+    isProgrammingExerciseResults = false;
+
     // Icons
     faDownload = faDownload;
 
     constructor(private resultService: ResultService, private alertService: AlertService) {}
 
+    ngOnInit(): void {
```

Comment:
GitHub seams to dislike this

Reply:
The problem was related to the unnecessary `Injectable` decorator, which I now have removed with https://github.com/ls1intum/Artemis/pull/5910/commits/e345facc762e63d1cc404d8e502ca7c64f8ef4ec 🙂